File: app/src/main/java/com/example/earthwallet/ui/nav/ActionsMainFragment.java
package com.example.earthwallet.ui.nav;

import com.example.earthwallet.R;
 
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.LinearLayout;
import android.widget.Toast;
import android.util.Log;
 
import androidx.fragment.app.Fragment;

public class ActionsMainFragment extends Fragment {

    private boolean isGovernanceExpanded = false;

    public ActionsMainFragment() {}
    
    public static ActionsMainFragment newInstance() {
        return new ActionsMainFragment();
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        return inflater.inflate(R.layout.fragment_actions_main, container, false);
    }

    @Override
    public void onViewCreated(View view, Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        
        // Swap Tokens button
        Button swapTokens = view.findViewById(R.id.btn_swap_tokens);
        if (swapTokens != null) {
            swapTokens.setOnClickListener(v -> {
                Log.d("ActionsMainFragment", "Swap Tokens button clicked");
                if (getActivity() instanceof com.example.earthwallet.ui.host.HostActivity) {
                    ((com.example.earthwallet.ui.host.HostActivity) getActivity()).showFragment("swap");
                } else {
                    Toast.makeText(getContext(), "Navigation not available", Toast.LENGTH_SHORT).show();
                }
            });
        }
    
        // ANML Claim button - request parent HostActivity to show ANML fragment
        Button anml = view.findViewById(R.id.btn_anml_claim);
        if (anml != null) {
            anml.setOnClickListener(v -> {
                Log.d("ActionsMainFragment", "ANML Claim button clicked");
                if (getActivity() instanceof com.example.earthwallet.ui.host.HostActivity) {
                    ((com.example.earthwallet.ui.host.HostActivity) getActivity()).showFragment("anml");
                } else {
                    Toast.makeText(getContext(), "Navigation not available", Toast.LENGTH_SHORT).show();
                }
            });
        }

        // Manage LP button - request parent HostActivity to show ManageLP fragment
        Button manageLP = view.findViewById(R.id.btn_manage_lp);
        if (manageLP != null) {
            manageLP.setOnClickListener(v -> {
                Log.d("ActionsMainFragment", "Manage LP button clicked");
                if (getActivity() instanceof com.example.earthwallet.ui.host.HostActivity) {
                    ((com.example.earthwallet.ui.host.HostActivity) getActivity()).showFragment("managelp");
                } else {
                    Toast.makeText(getContext(), "Navigation not available", Toast.LENGTH_SHORT).show();
                }
            });
        }

        // Stake ERTH button - request parent HostActivity to show Staking fragment
        Button stakeERTH = view.findViewById(R.id.btn_stake_erth);
        if (stakeERTH != null) {
            stakeERTH.setOnClickListener(v -> {
                Log.d("ActionsMainFragment", "Stake ERTH button clicked");
                if (getActivity() instanceof com.example.earthwallet.ui.host.HostActivity) {
                    ((com.example.earthwallet.ui.host.HostActivity) getActivity()).showFragment("staking");
                } else {
                    Toast.makeText(getContext(), "Navigation not available", Toast.LENGTH_SHORT).show();
                }
            });
        }

        // Governance button - toggle expansion of submenu
        Button governance = view.findViewById(R.id.btn_governance);
        LinearLayout governanceSubmenu = view.findViewById(R.id.governance_submenu);
        if (governance != null && governanceSubmenu != null) {
            governance.setOnClickListener(v -> {
                Log.d("ActionsMainFragment", "Governance button clicked");
                toggleGovernanceSubmenu(governanceSubmenu);
            });
        }

        // Caretaker Fund button
        Button caretakerFund = view.findViewById(R.id.btn_caretaker_fund);
        if (caretakerFund != null) {
            caretakerFund.setOnClickListener(v -> {
                Log.d("ActionsMainFragment", "Caretaker Fund button clicked");
                if (getActivity() instanceof com.example.earthwallet.ui.host.HostActivity) {
                    ((com.example.earthwallet.ui.host.HostActivity) getActivity()).showFragment("caretaker_fund");
                } else {
                    Toast.makeText(getContext(), "Navigation not available", Toast.LENGTH_SHORT).show();
                }
            });
        }

        // Deflation Fund button
        Button deflationFund = view.findViewById(R.id.btn_deflation_fund);
        if (deflationFund != null) {
            deflationFund.setOnClickListener(v -> {
                Log.d("ActionsMainFragment", "Deflation Fund button clicked");
                if (getActivity() instanceof com.example.earthwallet.ui.host.HostActivity) {
                    ((com.example.earthwallet.ui.host.HostActivity) getActivity()).showFragment("deflation_fund");
                } else {
                    Toast.makeText(getContext(), "Navigation not available", Toast.LENGTH_SHORT).show();
                }
            });
        }
    }
    
    private void toggleGovernanceSubmenu(LinearLayout submenu) {
        if (isGovernanceExpanded) {
            submenu.setVisibility(View.GONE);
            isGovernanceExpanded = false;
            Log.d("ActionsMainFragment", "Governance submenu collapsed");
        } else {
            submenu.setVisibility(View.VISIBLE);
            isGovernanceExpanded = true;
            Log.d("ActionsMainFragment", "Governance submenu expanded");
        }
    }
}

File: app/src/main/java/com/example/earthwallet/ui/nav/ActionsMainFragment.kt
package network.erth.wallet.ui.nav

import network.erth.wallet.ui.components.TxResult
import network.erth.wallet.R
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.LinearLayout
import android.widget.Toast
import androidx.fragment.app.Fragment

class ActionsMainFragment : Fragment() {

    private var isGovernanceExpanded = false

    companion object {
        @JvmStatic
        fun newInstance(): ActionsMainFragment = ActionsMainFragment()
    }

    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        return inflater.inflate(R.layout.fragment_actions_main, container, false)
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)

        // Swap Tokens
        view.findViewById<View>(R.id.btn_swap_tokens)?.setOnClickListener {
            (activity as? network.erth.wallet.ui.host.HostActivity)?.let {
                it.showFragment("swap")
            } ?: run {
                TxResult.message(requireContext(), "Couldn't continue", "Navigation not available")
            }
        }

        // ANML Claim
        view.findViewById<View>(R.id.btn_anml_claim)?.setOnClickListener {
            (activity as? network.erth.wallet.ui.host.HostActivity)?.let {
                it.showFragment("anml")
            } ?: run {
                TxResult.message(requireContext(), "Couldn't continue", "Navigation not available")
            }
        }

        // Manage LP
        view.findViewById<View>(R.id.btn_manage_lp)?.setOnClickListener {
            (activity as? network.erth.wallet.ui.host.HostActivity)?.let {
                it.showFragment("managelp")
            } ?: run {
                TxResult.message(requireContext(), "Couldn't continue", "Navigation not available")
            }
        }

        // Stake ERTH
        view.findViewById<View>(R.id.btn_stake_erth)?.setOnClickListener {
            (activity as? network.erth.wallet.ui.host.HostActivity)?.let {
                it.showFragment("staking")
            } ?: run {
                TxResult.message(requireContext(), "Couldn't continue", "Navigation not available")
            }
        }

        // Explorer (public chain data; needs no wallet)
        view.findViewById<View>(R.id.btn_explorer)?.setOnClickListener {
            (activity as? network.erth.wallet.ui.host.HostActivity)?.let {
                it.showFragment("explorer")
            } ?: run {
                TxResult.message(requireContext(), "Couldn't continue", "Navigation not available")
            }
        }

        // Governance - toggle expansion of submenu
        val governanceSubmenu = view.findViewById<LinearLayout>(R.id.governance_submenu)
        val governanceArrow = view.findViewById<ImageView>(R.id.governance_arrow)
        view.findViewById<View>(R.id.btn_governance)?.setOnClickListener {
            if (governanceSubmenu != null) {
                toggleGovernanceSubmenu(governanceSubmenu, governanceArrow)
            }
        }

        // Caretaker Fund (democratic, one-human-one-vote)
        view.findViewById<View>(R.id.btn_caretaker_fund)?.setOnClickListener {
            (activity as? network.erth.wallet.ui.host.HostActivity)?.let {
                it.showFragment("caretaker_fund")
            } ?: run {
                TxResult.message(requireContext(), "Couldn't continue", "Navigation not available")
            }
        }

        // Deflation Fund (stake-weighted)
        view.findViewById<View>(R.id.btn_deflation_fund)?.setOnClickListener {
            (activity as? network.erth.wallet.ui.host.HostActivity)?.let {
                it.showFragment("deflation_fund")
            } ?: run {
                TxResult.message(requireContext(), "Couldn't continue", "Navigation not available")
            }
        }
    }

    private fun toggleGovernanceSubmenu(submenu: LinearLayout, arrow: ImageView?) {
        if (isGovernanceExpanded) {
            submenu.visibility = View.GONE
            arrow?.rotation = 0f
            isGovernanceExpanded = false
        } else {
            submenu.visibility = View.VISIBLE
            arrow?.rotation = 90f
            isGovernanceExpanded = true
        }
    }
}